ui/app/ducks/swaps/swaps.js

```diff
@@ -50,6 +54,13 @@ const initialState = {
   quotesFetchStartTime: null,
   topAssets: {},
   toToken: null,
+  customGas: {
+    price: null,
+    limit: null,
+    loading: false,
```

Comment:
I think we have more than two states to consider here. There's the initial state, loading, completed, and failed.


If we fail to ever get gas price data (e.g. if the API is broken or something), it might be useful to represent that in the gas state. We could hide the gas buttons and only allow access to the gas input fields - something like that.

Reply:
Handling of all four states added in eed17631f